src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorEmitterTests.cs

```diff
@@ -163,6 +162,30 @@ internal static partial class TestWithDefaultValues
         }
 #endif
 
+        [Fact]
+        public async Task TestBaseline_TestWithNestedClassWithGenericTypesWithAttributes_Success()
```

Comment:
>TestBaseline_TestWithNestedClassWithGenericTypesWithAttributes_Success [](http://example.com/codeflow?start=26&length=70)

could you please add some test using the reflection to ensure the class GenericTypeWithAttribute type parameters still has the attributes Foo, Bar?